pyproject.toml

```diff
@@ -79,7 +79,7 @@ pydata-google-auth = { version = ">=1.4.0,<2", optional = true }
 pydruid = { version = ">=0.6.5,<1", optional = true, extras = ["sqlalchemy"] }
 pymysql = { version = ">=1,<2", optional = true }
 pyodbc = { version = ">=4.0.39,<6", optional = true }
-pyspark = { version = ">=3,<3.4", optional = true } # pyspark is heavily broken by numpy >=1.24 and pandas >=2
+pyspark = { version = ">=3", optional = true }
```

Comment:
Should we keep an upper bound on this at `<4`? PySpark does have a history of being incompatible with the latest pandas and numpy.

Reply:
Added.